lib/Dialect/FIRRTL/Transforms/ExpandWhens.cpp

```diff
@@ -152,16 +152,23 @@ class ExpandWhensVisitor : public FIRRTLVisitor<ExpandWhensVisitor> {
         condition.getLoc(), condition.getType(), condition, value);
   }
 
+  /// If a value has an outer flip, convert the value to passive.
+  Value convertToPassive(OpBuilder &builder, Location loc, Value input) {
+    auto inType = input.getType().cast<FIRRTLType>();
+    if (inType.isPassive())
+      return input;
+    return builder.create<AsPassivePrimOp>(loc, inType.getPassiveType(), input);
```

Comment:
createAndFold<AsPassivePrimOp> will handle the "already passive" case for you

Reply:
Thanks!